commit 7a985df6339499052bb7bad2df214477e7e1580a
Author: John Spray <john.spray@inktank.com>
Date:   Wed Feb 12 12:19:57 2014 +0000

    mds: Create MDSUtility as base for Dumper/Resetter
    
    These classes has lots of similar functionality and we may
    want to add more similar things in the future.
    
    Also, get a copy of the MDS map at startup
    to learn the ID of the metadata pool instead of
    assuming it's the default.
    
    Signed-off-by: John Spray <john.spray@inktank.com>

diff --git a/src/mds/MDSUtility.h b/src/mds/MDSUtility.h
new file mode 100644
index 00000000000..d3f938e8171
--- /dev/null
+++ b/src/mds/MDSUtility.h
@@ -0,0 +1,58 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2014 John Spray <john.spray@inktank.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ */
+
+#ifndef MDS_UTILITY_H_
+#define MDS_UTILITY_H_
+
+#include "osd/OSDMap.h"
+#include "osdc/Objecter.h"
+#include "mds/MDSMap.h"
+#include "messages/MMDSMap.h"
+#include "msg/Dispatcher.h"
+#include "msg/Messenger.h"
+#include "auth/Auth.h"
+
+/// MDS Utility
+/**
+ * This class is the parent for MDS utilities, i.e. classes that
+ * need access the objects belonging to the MDS without actually
+ * acting as an MDS daemon themselves.
+ */
+class MDSUtility : public Dispatcher {
+protected:
+  Objecter *objecter;
+  OSDMap *osdmap;
+  MDSMap *mdsmap;
+  Messenger *messenger;
+  MonClient *monc;
+
+  Mutex lock;
+  SafeTimer timer;
+
+  Context *waiting_for_mds_map;
+
+public:
+  MDSUtility();
+  ~MDSUtility();
+
+  void handle_mds_map(MMDSMap* m);
+  bool ms_dispatch(Message *m);
+  bool ms_handle_reset(Connection *con) { return false; }
+  void ms_handle_remote_reset(Connection *con) {}
+  bool ms_get_authorizer(int dest_type, AuthAuthorizer **authorizer,
+                         bool force_new);
+  int init();
+  void shutdown();
+};
+
+#endif /* MDS_UTILITY_H_ */
